### ISSUE TITLE
dev server unexpectedly loads CSS when using `?inline` imports

### ISSUE THREAD

**Issue by mayank99** [remix-run/remix#9844]
### Reproduction

https://stackblitz.com/edit/vitejs-vite-sfsfcx?file=styles.css&terminal=dev

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    @remix-run/dev: ^2.11.1 => 2.11.1 
    @remix-run/node: ^2.11.0 => 2.11.1 
    @remix-run/react: ^2.11.0 => 2.11.1 
    @remix-run/serve: ^2.11.0 => 2.11.1 
    vite: ^5.4.0 => 5.4.0
```

### Used Package Manager

npm

### Expected Behavior

Vite supports importing CSS strings without automatically injecting CSS into the page using an `?inline` query param (see [docs](https://vitejs.dev/guide/features.html#disabling-css-injection-into-the-page)).

```js
import rawCssText from './styles.css?inline'
```

This works as expected in vanilla Vite (see [comparison stackblitz](https://stackblitz.com/edit/vitejs-vite-cy4kgj?terminal=dev)) so it should also work fine in Remix.

### Actual Behavior

The CSS gets added to the page only when using the dev server (prod build works fine) and only on initial load (it gets removed after HMR).

Specifically, it gets added as a `<style>` tag inside the `<head>`.

**Comment by dj-rabel** [comment]
Experienced the same behavior in our project today and I think I know whats causing that.

https://github.com/remix-run/remix/blob/cc65962b1a96d1e134336aa9620ef1dad7c5efb1/packages/remix-dev/vite/styles.ts#L71-L75

There is a check for `?url`, but none for `?inline` and `?raw`. Thats why the initial serverside render includes the css.

**Comment by mayank99** [comment]
Just confirmed this is fixed in `2.12.1-pre.0` 👍
